### ISSUE TITLE
DamageOnlyWorkflow without itemCardId in options fails

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#797]
In GitLab by @napolitanod on Mar 31, 2022, 09:05

Hello!

I have been using midiQol damage only workflow to facilitate damage application in situations where damage is environmental and not associated with a particular item. I am seeing with a recent release (~ last week) that the damageOnlyWorkflow is failing and I believe that it is due to the options object that I am passing in as a parameter value not including an item (this is my guess). Is itemCardId now required and is there a more appropriate means that I should be using to apply damage when it doesn't come from an item?

The failure occurs at line 216 of utils.js line 216 (snippit below). I'm guessing the line needs a check for whether item is not null.

>>>
if (dmgTypeString !== "") {
		// if not checking all damage counts as magical
		let magicalDamage = item.data.data.properties?.mgc || item.data.flags.midiProperties?.magicdam;  <----------FAILURE IS HERE
		magicalDamage = magicalDamage || (configSettings.requireMagical === "off" && item?.data.data.attackBonus > 0);
		magicalDamage = magicalDamage || (configSettings.requireMagical === "off" && item?.data.type !== "weapon");
		magicalDamage = magicalDamage || (configSettings.requireMagical === "nonspell" && item?.data.type === "spell");
		const silverDamage = item?.data.data.properties?.sil;
>>>

crude example of what I actually am passing: MidiQOL.DamageOnlyWorkflow(null, null, damage.total, type, tokens, damage, {flavor: flavor})

**Comment by tposney** [comment]
Yes, known issue fix in 0.9.41. Thanks for the report

**Comment by tposney** [comment]
In GitLab by @napolitanod on Apr 1, 2022, 24:32

Thank you for the reply! I updated to 0.9.41 and I am no longer getting the error at the previous spot. I am now getting an error at line 157 in the workflow.js file. I'm not 100% I'm following the code at this point but I think at 157 it is expecting this.onUseMacros to have been populated with an object from line 154 that contains items and perhaps if no item is passed in with damageOnlyWorkflow this doesn't happen.

>>>
Danger Zone Spiked Wall Trap plan step execution failed undefined... TypeError: Cannot read properties of undefined (reading 'items')
    at new Workflow (modules/midi-qol/src/module/workflow.js:157)
    at new DamageOnlyWorkflow (modules/midi-qol/src/module/workflow.js:2503)
    at damageToken._applyDamage (modules/danger-zone/scripts/apps/workflow.js:1280)
>>>

>>>
	this.workflowOptions = options?.workflowOptions ?? {};
		if (configSettings.allowUseMacro) {
			this.onUseMacros = getProperty(this.item, "data.flags.midi-qol.onUseMacroParts");
			const actorOnUseMacros = getProperty(this.actor, "data.flags.midi-qol.onUseMacroParts");
			//@ts-ignore
			this.onUseMacros.items = (this.onUseMacros.items ?? []).concat(actorOnUseMacros.items); <--------ERROR IS HERE
		}
>>>

**Comment by tposney** [comment]
In GitLab by @ctbritt on Apr 1, 2022, 02:10

Getting a similar error when using ASE's version of magic missiles:
```
TypeError: Cannot read properties of undefined (reading 'items')
    at new Workflow (workflow.js:157:47)
    at new DamageOnlyWorkflow (workflow.js:2503:3)
    at MissileDialog._updateObject (missile-dialog.js:440:29)
    at async MissileDialog._onSubmit (foundry.js:3855:7)```

**Comment by tposney** [comment]
I am a dufus - apologies - fix in 0.9.42
